itest/send_test.go

```diff
@@ -1954,6 +1954,87 @@ func testSendNoCourierUniverseImport(t *harnessTest) {
 	AssertSendEventsComplete(t.t, receiveAddr.ScriptKey, sendEvents)
 }
 
+// testRestoreLndFromSeed tests that we can restore an LND node from a seed and
+// then continue to interact with assets previously minted on the node.
+func testRestoreLndFromSeed(t *harnessTest) {
+	// We create a new lnd node from a seed, so we can restore it with the
+	// same seed later.
+	password := []byte("somepassword")
+	seedLnd, mnemonic, _ := t.lndHarness.NewNodeWithSeed(
+		"seed-lnd", lndDefaultArgs, password, false,
+	)
+	t.lndHarness.FundCoins(btcutil.SatoshiPerBitcoin, seedLnd)
+
+	bob := setupTapdHarness(t.t, t, seedLnd, t.universeServer)
+	defer func() {
+		require.NoError(t.t, bob.stop(!*noDelete))
+	}()
+
+	// We mint a batch of normal assets with enough units to allow us to
+	// send it around a few times.
+	rpcAssets := MintAssetsConfirmBatch(
+		t.t, t.lndHarness.Miner().Client, bob,
+		[]*mintrpc.MintAssetRequest{issuableAssets[0]},
+	)
+
+	var (
+		alice    = t.tapd
+		ctxb     = context.Background()
+		rpcAsset = rpcAssets[0]
+		genInfo  = rpcAsset.AssetGenesis
+	)
+
+	// We send some of the minted assets to our default tapd node.
+	const sendAmount = 123
+	aliceAddr, err := alice.NewAddr(ctxb, &taprpc.NewAddrRequest{
+		AssetId: genInfo.AssetId,
+		Amt:     sendAmount,
+	})
+	require.NoError(t.t, err)
+
+	AssertAddrCreated(t.t, alice, rpcAsset, aliceAddr)
+
+	sendResp, sendEvents := sendAssetsToAddr(t, bob, aliceAddr)
+
+	ConfirmAndAssertOutboundTransfer(
+		t.t, t.lndHarness.Miner().Client, bob, sendResp,
+		genInfo.AssetId,
+		[]uint64{rpcAsset.Amount - sendAmount, sendAmount}, 0, 1,
+	)
+	AssertNonInteractiveRecvComplete(t.t, alice, 1)
+	AssertSendEventsComplete(t.t, aliceAddr.ScriptKey, sendEvents)
+
+	// We now restore Bob's lnd node from the seed.
+	require.NoError(t.t, bob.stop(false))
+	require.NoError(t.t, seedLnd.Shutdown())
+
+	// Starting the node again should restore it to the same state as
+	// before. This takes a couple of seconds, so let's log that we're
+	// waiting for the node to start.
+	t.Logf("Restoring node from seed, this may take a few seconds...")
+	seedLnd = t.lndHarness.RestoreNodeWithSeed(
+		"lnd-seed-restored", lndDefaultArgs, password, mnemonic, "",
+		2500, nil,
+	)
+	require.NoError(t.t, updateConfigWithNode(bob.clientCfg, seedLnd))
+
+	require.NoError(t.t, bob.start(false))
+
+	// Send more assets after restoring the node.
+	aliceAddr, err = alice.NewAddr(ctxb, &taprpc.NewAddrRequest{
+		AssetId: genInfo.AssetId,
+		Amt:     sendAmount,
+	})
+	require.NoError(t.t, err)
+
+	AssertAddrCreated(t.t, alice, rpcAsset, aliceAddr)
+
+	sendAsset(
```

Comment:
So when this fails, it's because the wrong key was derived? Is it possibly us (`tapd`) passing in the incorrect information? Like an issue in a db query somewhere or something.

Reply:
Yeah, the wrong key is derived. The default case is family 0 and index 0 if the key isn't found by pubkey only.
So nothing wrong in `tapd`, just weird `lndclient` behavior.